### DISCUSSION TITLE
What's the best way to add/append a new IPSec policy?

### DISCUSSION THREAD

**Discussion by JimChen111** [strongswan/strongswan#2413] [Q&A] [answered]
Hi,

Currently, there are some existing IPSec policies in my `swanctl.conf`. If I want to dynamically add some more policies, what's the best way to do it? It seems if I use a new config file and load it with `swanctl --load-all`, the old policies will be overriden. So I suppose we have to add the new policy into the same config file and then reload it? But AFAIK, it's not very intuitive to edit the `swanctl.conf` file through programming since we need to manually parse it?

Thanks.

**Comment by strongX509** [comment, marked as answer]
Instead of the **`swanctl`** command line tool which parses the **`swanctl.conf`** file you could directly access the [VICI interface](https://github.com/strongswan/strongswan/tree/master/src/libcharon/plugins/vici#readme) where APIs for Python, Ruby, Perl and GoLang are available.

**Comment by tobiasbrunner** [comment]
> it seems if I use a new config file and load it with `swanctl --load-all`, the old policies will be overriden.

What do you mean?

> But AFAIK, it's not very intuitive to edit the `swanctl.conf` file through programming since we need to manually parse it?

Parse? Also, includes are supported, so you don't necessarily have to edit any files, but could just remove/add separate files for separate connections.
